Fix ExecutionUpdatableTask attempt state tracking to record proper duration

KarthikReddyMaru opened the pull request:
This PR ensures that ExecutionUpdatableTask attempts properly record state transitions (**CREATED → RUNNING → SUCCESS**) with accurate timestamps, enabling correct duration tracking and complete attempt history visibility in the UI.

closes #12177 

## What changes are being made and why?

The `handleExecutionUpdatingTask` method was creating attempt records with only the CREATED and terminal states, skipping the RUNNING state transition. This caused the **RUNNING state to be missing** from attempt history, resulting in incomplete execution timeline in UI.

The fix updates the same attempt object through complete state transitions:

1. Before task execution: transition attempt from **CREATED → RUNNING**
2. After task execution: transition attempt from **RUNNING → terminal state**

## How the changes have been QAed?

**Before** 

<img width="1217" height="704" alt="Screenshot 2025-10-21 at 6 15 11 PM" src="https://github.com/user-attachments/assets/f71bc81d-3dfd-4d86-8ff6-81e4f156a226" />

*Missing RUNNING state: Only CREATED and SUCCESS states visible*

**After**

<img width="1217" height="704" alt="Screenshot 2025-10-21 at 6 13 04 PM" src="https://github.com/user-attachments/assets/847f5eae-94fb-4f55-9003-7d1bd4c8ca87" />

*Complete state progression: CREATED → RUNNING → SUCCESS with distinct timestamps*

**Verification:**

- Complete state progression: CREATED → RUNNING → terminal
- UI displays proper attempt timeline
- No regression for skipped tasks (CREATED → SKIPPED)

KarthikReddyMaru committed:
fix(executor): record CREATED/RUNNING/SUCCESS states for ExecutionUpdatableTask attempts

Previously, ExecutionUpdatableTask attempts were created with only the
terminal state, skipping RUNNING and causing zero duration and incorrect UI display.

Update attempts through all state transitions:
- CREATED → RUNNING before task.update() call
- RUNNING → terminal state after execution

This preserves complete state history and enables accurate duration tracking.

loicmathieu commented on executor/src/main/java/io/kestra/executor/ExecutorService.java:
```diff
@@ -1083,19 +1081,43 @@ private Executor handleExecutionUpdatingTask(final Executor executor) throws Int
+                        .getTaskRun()
+                        .withAttempts(attempts)
+                        .withState(State.Type.RUNNING);
+
```
At this stage, there cannot be any attemps so I think this could be simplified to

```java
TaskRun runningTaskRun = workerTask
                        .getTaskRun()
                        .withAttempts(List.of(TaskRunAttempt.builder().state(new State().withState(State.Type.RUNNING)).build())
                        .withState(State.Type.RUNNING);
```

loicmathieu reviewed (changes requested):
Can you also add assertions into an existing test, for ex the `RuntimeLabelsTest`?

KarthikReddyMaru commented:
Sure, I’ll add the assertions to RuntimeLabelsTest and update the PR shortly.

KarthikReddyMaru committed:
refactor: simplify attempt creation in handleExecutionUpdatingTask

Remove null/empty check for attempts list as no attempts exist at this
execution stage. Create attempt directly with RUNNING state using
`new State().withState(State.Type.RUNNING)`.

This simplifies the code while maintaining correct state transitions
(CREATED → RUNNING → terminal).

KarthikReddyMaru committed:
test(RuntimeLabels): add state transition assertions
for ExecutionUpdatableTask in Labels tests

Verify that ExecutionUpdatableTask attempts record complete state transitions
(CREATED → RUNNING → SUCCESS) across all Labels test scenarios

loicmathieu reviewed (changes requested):
LGTM, thanks
